File: frontapp/src/model/post.ts
import { Comment } from './comment';
import { Cause } from './common';
import { Member } from './member';

export interface Post {
  id: number;
  title: string;
  subtitle: string;
  content: string;
  isPublic: boolean;
  isPaid: boolean;
  createDate: string;
  modifyDate: string;
  writerId: number;
  writerUsername: string;
  comments?: Comment[];
  cause?: Cause;
}

export interface WritePost
  extends Pick<
    Post,
    'title' | 'subtitle' | 'content' | 'isPublic' | 'isPaid'
  > {}

File: src/main/java/com/ll/medium/domain/post/post/entity/Post.java
package com.ll.medium.domain.post.post.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.ll.medium.domain.member.member.entity.Member;
import com.ll.medium.domain.post.comment.entity.Comment;
import com.ll.medium.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@SuperBuilder(toBuilder = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class Post extends BaseEntity {
    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String subtitle;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    private boolean isPublic = true;

    private boolean isPaid = false;

    //@JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "writer_id", nullable = false)
    private Member writer;

    //@JsonIgnore
    @OneToMany(mappedBy = "post", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Comment> comments = new ArrayList<>();

    public boolean getIsPublic() {
        return this.isPublic;
    }

    public boolean getIsPaid() {
        return this.isPaid;
    }
}
